osc-server/src/test/java/org/osc/core/broker/util/crypto/X509TrustManagerFactoryTest.java

```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.util.crypto;
+
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.osc.core.broker.service.api.server.ArchiveApi;
+import org.osc.core.broker.service.api.server.LoggingApi;
+import org.osc.core.broker.service.archive.ArchiveUtil;
+
+public class X509TrustManagerFactoryTest {
+
+    private static final String TEST_CERT_FILE_NAME = "testcertificate.crt";
+    private static final String TEST_TRUSTSTORE_FILE_NAME = "osctrustore.jks";
+    private static final String TEST_CERT_ALIAS = "testcertificate";
+
+    private static final String TEST_X509PEM_ZIP_FILE = "oscx509test_x509pem.zip";
+    private static final String TEST_PKICHAIN_ZIP_FILE_NAME = "oscx509test.zip";
+    private static final String TEST_PRIVATE_KEY_FILE_NAME = "oscx509test.pem";
+
+    private File testCertFile;
+    private File testTrustStoreFile;
+    private File testPKIChainZipFile;
+    private File testX509PEMZipFile;
+    private File testPrivateKeyFile;
+
+    @Mock
+    private LoggingApi loggingApi;
+
+    @Spy
+    @InjectMocks
+    private ArchiveApi archiveUtil = new ArchiveUtil();
```

Comment:
do we need this? doesnt seem like we are setting up anything with the archive util